src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java

```diff
@@ -319,11 +357,50 @@ public String getString2(final String component,
                     LocaleUtility.defaultLocale);
             source = rb.getString(key);
         } catch (Exception e) {
-            this.logger.error(e.getMessage());
+            logger.error(e.getMessage());
+            source = key;
+        }
+        // get translation from VIP service
+        message = getString(locale, component, key, source, "", args);
+        return message;
+    }
+
+    /**
+     * get one translation of the configured product from VIP, if message not
+     * found will get the English message from specified bundle.
+     *
+     * @param component
+     *            defined on VIP service, it will be created automatically if
+     *            not exist
+     * @param bundle
+     *            properties file name, normally it should be put under the root
+     *            'src' path
+     * @param locale
+     *            an object used to get the source's translation
+     * @param key
+     *            identify the source
+     * @param args
+     *            named arguments used to format the message with placeholder
+     * @return a message of translation, if the translation is not found from
+     *         VIP service, it will return the value defined in the bundle
+     *         searching by the key
+     */
+    public String getString2(final String component, final String bundle, final Locale locale, final String key,
+            final Map<String, Object> arguments) {
+        logger.trace("Start to execute TranslationMessage.getStringWithNamedArgs");
+        if (key == null || key.equalsIgnoreCase(""))
+            return "";
+        String message = "";
+        String source;
+        try {
+            ResourceBundle rb = ResourceBundle.getBundle(bundle, LocaleUtility.defaultLocale);
+            source = rb.getString(key);
+        } catch (Exception e) {
+            logger.error(e.getMessage());
             source = key;
         }
         // get translation from VIP service
```

Comment:
Can we have another private method for these codes and both getString2 can share it?